app/assets/javascripts/mtl/document_modal.coffee

```diff
@@ -0,0 +1,105 @@
+IMAGE_EXTENSIONS = ['png', 'jpg', 'gif', 'bmp']
+MTL.templates.document_modal = """
+  <div class="document-modal">
+    <div class="document-modal-dimmer"></div>
+    <div class="document-modal-header">
+      <div class="document-modal-title left white-text">
+        <%= MTL.fileIcon(filename, { class: 'left' }) %>
+        <%- filename %>
+      </div>
+      <div class="document-modal-toolbar right">
+        <a href="<%- url %>" target="_blank">
+          <%= MTL.icon('file_download', { class: 'white-text left' }) %>
+        </a>
+        <a href="#" data-mtl-document-modal="close">
+          <%= MTL.icon('close', { class: 'white-text left' }) %>
+        </a>
+      </div>
+
+      <div class="document-modal-content">
+      </div>
+
+      <% if(hasNext) { %>
+        <div class="document-modal-next btn black">
+          <%= MTL.icon('keyboard_arrow_right', { class: 'white-text' }) %>
+        </div>
+      <% } %>
+
+      <% if(hasPrev) { %>
+        <div class="document-modal-prev btn black">
+          <%= MTL.icon('keyboard_arrow_left', { class: 'white-text' }) %>
+        </div>
+      <% } %>
+    </div>
+  </div>
+"""
+
+MTL.templates.preview_unknown = """
+  <div class="no-preview">
+    <p class="center">
+      <%= MTL.icon('visibility_off', { size: 'large' }) %>
+    </p>
+    <a href="<%- url %>" target="_blank" class="btn btn-primary truncate">
+      <%= MTL.icon('file_download', { class: 'left' }) %>
+      <%- filename %>
+    </a>
+  </div>
+"""
+
+nextFileCard = ($fileCard) ->
+  $fileCard.next('[data-mtl-document-modal="open"]')
+
+prevFileCard = ($fileCard) ->
+  $fileCard.prev('[data-mtl-document-modal="open"]')
+
+hasNextFileCard = ($fileCard) ->
+  nextFileCard($fileCard).length > 0
+
+hasPrevFileCard = ($fileCard) ->
+  prevFileCard($fileCard).length > 0
+
+fileExtension = (filename) ->
+  String(filename).toLowerCase().split('.').pop()
+
+isImage = (filename) ->
+  _.includes(IMAGE_EXTENSIONS, fileExtension(filename))
+
+isPDF = (filename) ->
+  fileExtension(filename) == 'pdf'
+
+createPreview = ($container, filename, url) ->
+  switch
+    when isImage(filename)
+      $container.append($('<img/>', { src: url, alt: filename }))
+    when isPDF(filename) && PDFObject.supportsPDFs
+      PDFObject.embed(url, $container, { width: '800px', height: $(window).height() + 'px' })
```

Comment:
this is a pita... normally PDFObject adds an inline height of 100% but this doesn't work since the parent container `document-modal-content` stretches with its children (i.e has no fixed height)